### PR TITLE
Add segment span attributes

### PR DESCRIPTION
**Pull request by michaelgoin**
## Links

* Closes: https://github.com/newrelic/newrelic-node-apollo-server-plugin/issues/20

## Details

Adds an initial set of operation and field resolve level attributes.

Field args are initially set to a destination of NONE so they do not show up unless explicitly enabled via attribute config include. We'll still want to discuss with security so they are aware and confident we are doing the right thing (we are).

Attribute names subject to change.

### CODE REVIEW

File: lib/create-plugin.js [hunk 1]
```diff
@@ -65,9 +81,24 @@ function createPlugin(instrumentationApi) {
...
+                deepestResolvedPath.depth = pathArray.length
+                deepestResolvedPath.formatted = formattedPath
+              }
+              resolverSegment.addAttribute(FIELD_PATH_ATTR, formattedPath)
+              resolverSegment.addAttribute(FIELD_NAME_ATTR, info.fieldName)
+              resolverSegment.addAttribute(RETURN_TYPE_ATTR, info.returnType.toString())
+              resolverSegment.addAttribute(PARENT_TYPE_ATTR, info.parentType.toString())
+
+
+              for (const [key, value] of Object.entries(args)) {
```
Comment (carlo-808) [inline lib/create-plugin.js:94]: Do you think there's value is adding args as attribute without a 'value'?

File: tests/versioned/agent-testing.js [hunk 2]
```diff
@@ -31,8 +31,25 @@ function findSpanById(agent, spanId) {
   return matchingSpan
 }
 
+function findSegmentByName(root, name) {
```
Comment (carlo-808) [inline tests/versioned/agent-testing.js:34]: nice helper func!

**Review by carlo-808 — APPROVED** [review]
I see attributes added for resolver segments with relevant tests for apollo-server, express and hapi.  👍

File: lib/create-plugin.js (hunk 1 above)
Comment (michaelgoin) [reply lib/create-plugin.js:94]: I considered that but ultimately decided to keep it simple and follow a similar model to request params. We can always look to add later if we find there's a need.

**Pull request merged**